src/network/netdev/wireguard.c

```diff
@@ -846,38 +846,35 @@ static void wireguard_done(NetDev *netdev) {
         set_free(w->peers_with_failed_endpoint);
 }
 
-static int wireguard_read_private_key_file(Wireguard *w, bool fatal) {
+static int wireguard_read_private_key_file(Wireguard *w) {
         _cleanup_free_ char *contents = NULL;
         _cleanup_free_ void *key = NULL;
         size_t size, key_len;
         NetDev *netdev;
-        int level, r;
+        int r;
 
         assert(w);
-
         netdev = NETDEV(w);
+        assert(netdev);
 
         if (!w->private_key_file)
                 return 0;
 
-        level = fatal ? LOG_ERR : LOG_INFO;
-
         r = read_full_file(w->private_key_file, &contents, &size);
         if (r < 0)
-                return log_netdev_full(netdev, level, r,
-                                       "Failed to read private key from '%s'%s: %m",
-                                       w->private_key_file, fatal ? "" : ", ignoring");
+                return log_netdev_error_errno(netdev, r,
+                                              "Failed to read private key from '%s', ignoring: %m",
+                                              w->private_key_file);
 
         r = unbase64mem(contents, size, &key, &key_len);
         if (r < 0)
-                return log_netdev_full(netdev, level, r,
-                                       "Failed to decode private key%s: %m",
-                                       fatal ? "" : ", ignoring");
+                return log_netdev_error_errno(netdev, r,
+                                              "Failed to decode private key, ignoring: %m");
 
         if (key_len != WG_KEY_LEN)
-                return log_netdev_full(netdev, level, SYNTHETIC_ERRNO(EINVAL),
-                                       "Wireguard private key has invalid length (%zu bytes)%s: %m",
-                                       key_len, fatal ? "" : ", ignoring");
+                return log_netdev_error_errno(netdev, SYNTHETIC_ERRNO(EINVAL),
```

Comment:
(while we are at it: actually i think it would make a ton of sense to use explicit_bzero_safe(key, key_len) here to clear out the private key again...)